### PR TITLE
Force use of legacy obw to ensure woo pages are setup.

### LINKED ISSUE #522: Remove old Onboarding Code
As @joshuatf [pointed out here](https://github.com/Automattic/wc-calypso-bridge/pull/521#issuecomment-596502412) we have some clean up to do to remove some old code.

### PR DESCRIPTION
**Pull request by timmyc**
Fixes #522 
Fixes https://github.com/Automattic/wp-calypso/issues/39850

Currently new users coming through from the calypso setup flow to the ecommerce plan can get opted-in to the new store profiler experience, but a shortened one. This can result in a site being "setup" without the default Woo pages ( shop, cart, checkout ) from ever being created.

This branch seeks to fix that bug by forcing e-comm users to use the modified "legacy" OBW that originally shipped with the project. The legacy OBW [creates the pages](https://github.com/woocommerce/woocommerce/blob/master/includes/admin/class-wc-admin-setup-wizard.php#L770) during this flow, and thus fixes the bug reported in Calypso.

Additionally in this PR I removed other code that @joshuatf pointed out that no longer is needed since we are forcing users to go through the legacy OBW and Calypsoify checklist.

Lastly, note that all of this code will go away when we remove Calypsoify from the ecomm plan woo side of things :)

__To Test__
On `master` of calypso-bridge

- Install WooCommerce 4.0
- Opt in to the new onboarding experience by `update_option( 'woocommerce_setup_ab_wc_admin_onboarding', 'b' )`
- Also handy to add this filter to always ensure the calypsoify setup checklist is shown `add_filter( 'wc_calypso_bridge_development_mode', '__return_true' );`
- Visit `wp-admin?page=wc-setup` verify that you are shown the option to use the new onboarding experience with the "YES PLEASE" button.

Install this branch

- Visit `wp-admin?page=wc-setup`, and verify you see the following flow:

wc-calypso-bridge1
WooCommerce › Setup Wizard - Mozilla Firefox 2020-03-12 13-26-40
Setup ‹ ecomm woo test — WordPress - Mozilla Firefox 2020-03-12 13-26-59

### CODE REVIEW

**Comment by octaedro** [comment]
Hi @timmyc the code looks really good.
I created a new site and followed all the mentioned steps. However, I don't know if I'm having some local problem but after pressing `Yes please`
screenshot-store-setup0

A message appears saying: `Welcome to WooCommerce! ...`
screenshot-store-setup

When I press `Continue Store Setup`, it reloads the page. Behind the scenes it's sending me to `/wp-admin/admin.php?page=wc-admin&enable_onboarding=1` and from there I'm receiving a redirect (`Status Code: 302 Found`) to `/wp-admin/admin.php?page=wc-admin`

Do you have any idea about what could be going on?

File: includes/class-wc-calypso-bridge-woocommerce-admin.php [hunk 1]
```diff
@@ -37,6 +37,7 @@ public function init() {
 		add_filter( 'wc_admin_get_feature_config', array( $this, 'maybe_remove_devdocs_menu_item' ) );
 		add_filter( 'pre_option_woocommerce_task_list_hidden', array( $this, 'disable_new_task_list' ) );
 		add_filter( 'pre_option_woocommerce_onboarding_opt_in', array( $this, 'disable_onboarding_opt_in' ) );
+		add_filter( 'pre_option_woocommerce_setup_ab_wc_admin_onboarding', array( $this, 'disable_onboarding_a_b_test' ) );
```
Comment (joshuatf) [inline includes/class-wc-calypso-bridge-woocommerce-admin.php]: If we don't have the `new_onboarding` step we hopefully won't hit this check at all.

https://github.com/woocommerce/woocommerce/blob/6a8d8b27d98fe3036f2240d14be2b7a5e4ecf10e/includes/admin/class-wc-admin-setup-wizard.php#L278

**Comment by joshuatf** [comment]
@octaedro This plugin's calypsoify methods may not be active since that redirect comes from the new onboarding which should be disabled by this plugin.

Can you try appending `calypsoify=1` to the end of your URL and see if you still get redirected?

**Comment by octaedro** [comment]
> Can you try appending calypsoify=1 to the end of your URL and see if you still get redirected?

@joshuatf I'm adding `calypsoify=1` at the end of the redirect URL but it's still the same result  :confused:

**Comment by timmyc** [comment]
Okay I pushed some updates from your feedback @joshuatf - thanks for that!

@octaedro interesting so if you go to `wp-admin?page=wc-setup` you get redirected? Give me a ping when you want to review this together and I can help check it out.

**Comment by octaedro** [comment]
> @octaedro interesting so if you go to `wp-admin?page=wc-setup` you get redirected? Give me a ping when you want to review this together and I can help check it out.

I found what my problem was: my branch :sweat_smile:
Now it's working well!

**Comment by joshuatf** [comment]
Just saw @octaedro's review; looks like we're good here!  :shipit:

**Comment by timmyc** [comment]
Thanks to you both @octaedro and @joshuatf !!

**Pull request merged**